Move iPadMini UA overrides to Quirks.cpp

karlcow opened the pull request:
#### c0bc71135b421a7ed16cd9b2e76c47a4954f806a
<pre>
Move iPadMini UA overrides to Quirks.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=269335">https://bugs.webkit.org/show_bug.cgi?id=269335</a>
<a href="https://rdar.apple.com/122920694">rdar://122920694</a>

Reviewed by Wenson Hsieh.

Currently the iPad UA override are handled in
Source/WebKit/UIProcess/iOS/WebPageProxyIOS.mm
The Quirks are usually handled in Source/WebCore/page/Quirks.cpp

It creates two separate places for checking the domain names when
there&apos;s a website with a need for a specific treatment.
Moving the domain name check in the same place than the rest of the
Quirks will make it easier for people new to code to understand all
the deviation from the normal behavior.

It doesn&apos;t change the current behavior. So when a domain name is matching
the userAgent will be set to RecommendDesktopClassBrowsingForRequest::No
which is basically the iPadOS (aka iPad mini) user agent.

Note that this will not solve the needsQuirk check.
It requires a bigger refactor for checking if Quirks are active or not.
It should be a good next step.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsIpadMiniUserAgent const):
* Source/WebCore/page/Quirks.h:
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
(WebKit::desktopClassBrowsingRecommendedForRequest):

Canonical link: <a href="https://commits.webkit.org/274631@main">https://commits.webkit.org/274631@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9975f47db839cbedd58f82a2f3155f212442028e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/39570 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/18549 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41926 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/42105 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/35470 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/41876 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/21433 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15878 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/33046 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40144 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/15627 "Passed tests") | [⏳ 🧪 api-mac](https://ews-build.webkit.org/#/builders/API-Tests-macOS-EWS "Waiting to run tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/13566 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/13540 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/35197 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/43382 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35917 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/35528 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/39327 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/14382 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11845 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/37590 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15988 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8877 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/16036 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/15645 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebCore/page/Quirks.h:
```diff
@@ -102,6 +102,7 @@ class Quirks {
 
     WEBCORE_EXPORT static void updateStorageAccessUserAgentStringQuirks(HashMap<RegistrableDomain, String>&&);
     WEBCORE_EXPORT String storageAccessUserAgentStringQuirkForDomain(const URL&);
+    WEBCORE_EXPORT bool needsIpadMiniUserAgent(auto host)  const;
```
```suggestion
    WEBCORE_EXPORT static bool needsIpadMiniUserAgent(StringView host);
```

karlcow replied:
Thanks!

whsieh commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -1728,4 +1728,88 @@ bool Quirks::shouldDisableNavigatorStandaloneQuirk() const
+// FIXME: find the reference radars and/or bugs.webkit.org issues on why these were added in the first place.
+// FIXME: There is no check currently on needsQuirks(), this needs to be fixed so it makes it easier
+// to deactivate them for testing.
+bool Quirks::needsIpadMiniUserAgent(auto host)  const
```
```suggestion
bool Quirks::needsIpadMiniUserAgent(StringView host)
```

karlcow replied:
I changed `auto` by `StringView`

whsieh reviewed (approved):
r=mews

karlcow committed:
Move iPadMini UA overrides to Quirks.cpp
https://bugs.webkit.org/show_bug.cgi?id=269335
rdar://122920694

Reviewed by Wenson Hsieh.

Currently the iPad UA override are handled in
Source/WebKit/UIProcess/iOS/WebPageProxyIOS.mm
The Quirks are usually handled in Source/WebCore/page/Quirks.cpp

It creates two separate places for checking the domain names when
there's a website with a need for a specific treatment.
Moving the domain name check in the same place than the rest of the
Quirks will make it easier for people new to code to understand all
the deviation from the normal behavior.

It doesn't change the current behavior. So when a domain name is matching
the userAgent will be set to RecommendDesktopClassBrowsingForRequest::No
which is basically the iPadOS (aka iPad mini) user agent.

Note that this will not solve the needsQuirk check.
It requires a bigger refactor for checking if Quirks are active or not.
It should be a good next step.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsIpadMiniUserAgent const):
* Source/WebCore/page/Quirks.h:
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
(WebKit::desktopClassBrowsingRecommendedForRequest):

Canonical link: https://commits.webkit.org/274631@main
